compiler/src/dotty/tools/dotc/core/tasty/TreeUnpickler.scala

```diff
@@ -702,6 +702,10 @@ class TreeUnpickler(reader: TastyReader, nameAtRef: NameRef => TermName, posUnpi
       if (!sym.isType) { // Only terms might have leaky aliases, see the documentation of `checkNoPrivateLeaks`
         sym.info = ta.avoidPrivateLeaks(sym, tree.pos)
       }
+      if ((sym.isClass || sym.is(CaseVal)) && sym.isLocal)
```

Comment:
Why not use `isLocal` instead of `isInaccessibleChildOf` here?

Reply:
Because we don't have a parent yet against which to test with `inAccessibleChildOf`